### PR TITLE
Fixes #476 Printing a patient record

### LINKED ISSUE #476: Printing a patient record
The standard, out-of-the-box printed version of a patient record, including their history needs to be added to the primary patient summary screen, perhaps as a button opposite to the "Add Note" action @jglovier?

screen shot 2016-05-22 at 2 58 49 pm

For starters, the printed report should involve:

- [ ] A standard header that includes customizable language around the fact that this report is for authorized person(s) only
- [ ] A printable version of the upper section "patient summary" fields (as shown in the screenshot above)
- [ ] A list of the patient history, streatching as many pages as needed

### PR DESCRIPTION
**Pull request by dapierce**
Fixes #476 Printing a patient record

- Adds print stylesheet, with printing-specific css.
- Removed left padding of navigation from the print style, and set basic color and margin styles for printing.
- Adds definition of a new css class ‘print-section’, which will only show on a printed report.
- In app/locales/en/translations.js, added the string 'messages.for_authorized_persons' with the message: 'This report is for authorized persons only.' (Will require actual translation into additional languages, in this commit I just used Google Translate for placeholder translations into all the current language files).
- Added a header to app/templates/section.hbs to add messages.for_authorized_persons text, using the ‘print-section’ class.

This might not be exactly as envisioned, but I would appreciate any feedback. Thanks!

cc @HospitalRun/core-maintainers

### CODE REVIEW

**Comment by jkleinsc** [comment]
Thanks for the pr @dapierce !  We will take a look at this.  @tangollama what do you think?  @jglovier any issues/concerns about the css changes here?

**Comment by jglovier** [comment]
@dapierce can you please add an image of the final outcome here for quick reference?

**Comment by dapierce** [comment]
Here are a couple examples:

Patient page as seen in Print dialog box

Viewing same page in Chrome dev tools emulating print media

**Comment by jglovier** [comment]
There a couple things we could clean up later, but in general this looks great and is super helpful. :metal: 👍 :zap:

🚢

File: app/styles/_print.scss [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    background: white !important;
+    color: black !important;
+    font-size: 12pt;
+    margin: 0 5%;
+    &.ember-application { padding-left: 0; }
+  }
+
+  .print-section { display: block; text-align: center; }
+
+  .panel-footer, .btn, .nav, .view-current-title { display: none; }
```
Comment (jglovier) [inline app/styles/_print.scss]: Each selector should be on their own line. In fact I thought we had a test for that in the Stylelint. Wonder why it's not failing the build? cc @billybonks

File: app/styles/_print.scss [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+
+  body {
+    background: white !important;
+    color: black !important;
+    font-size: 12pt;
+    margin: 0 5%;
+    &.ember-application { padding-left: 0; }
+  }
+
+  .print-section { display: block; text-align: center; }
```
Comment (jglovier) [inline app/styles/_print.scss]: Same here: those declarations should each be on a new line.

File: app/styles/_print.scss [hunk 3]
```diff
@@ -0,0 +1,19 @@
+.print-section { display: none; }
+
+@media print {
+
+  body {
+    background: white !important;
+    color: black !important;
+    font-size: 12pt;
+    margin: 0 5%;
```
Comment (jglovier) [inline app/styles/_print.scss]: Selector order is wrong in this block.

File: app/styles/_print.scss [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+    &.ember-application { padding-left: 0; }
+  }
+
+  .print-section { display: block; text-align: center; }
+
+  .panel-footer, .btn, .nav, .view-current-title { display: none; }
+
+  .patient-history-heading { border-bottom: 1px solid black; }
```
Comment (jglovier) [inline app/styles/_print.scss]: Should only be using a variable, not a color keyword.

**Comment by jglovier** [comment]
Found some CSS syntax issues that need cleaned up before merging. Wonder why Stylelint isn't failling the build on these though?

**Comment by jkleinsc** [comment]
@jglovier Not sure why stylelint isn't failing the build here, but if I test this code change with the latest from master, it does throw an error:
```
not ok 7 PhantomJS 2.1 - Style Lint: _print.scss should pass stylelint
    ---
        actual: >
            false
        expected: >
            true
        stack: >
            http://localhost:7357/assets/test-support.js:4130:17
            http://localhost:7357/assets/tests.js:54:7
            runTest@http://localhost:7357/assets/test-support.js:2779:32
            run@http://localhost:7357/assets/test-support.js:2764:11
            http://localhost:7357/assets/test-support.js:2906:14
            process@http://localhost:7357/assets/test-support.js:2565:24
            begin@http://localhost:7357/assets/test-support.js:2547:9
            http://localhost:7357/assets/test-support.js:2607:9
        message: >
            7:5 Expected variable for "color". (sh-waqar/declaration-use-variable)
        Log: |
    ...
```

@dapierce can you correct this issue?  We should be able to accept the PR once this is set.

**Comment by billybonks** [comment]
@jkleinsc because he is using a colour that hasn't been set to a variable 😄

**Comment by dapierce** [comment]
I will change to color variables in the print scss--been away for a bit, but I should be able to update the PR this week.

**Comment by billybonks** [comment]
@dapierce i love the fact that you added translations into every locale 👍 .

**Comment by dapierce** [comment]
I have updated my PR, but I'm not really sure why the tests are all failing on this one.

File: app/styles/_print.scss [hunk 5]
```diff
@@ -9,8 +9,8 @@
   }
 
   body {
-    font-size: 12pt;
     margin: 0 5%;
+    font-size: 12pt;
```
Comment (jglovier) [inline app/styles/_print.scss]: Should use `px` values here.

**Comment by jglovier** [comment]
@dapierce btw for your convenience, if you are using Atom there is a handy linter package you can install that will show you SCSS linting failures in real time as you work, and what the fix is. [More on that here](https://github.com/HospitalRun/hospitalrun-frontend#the-scss-linter).

**Comment by dapierce** [comment]
I have made updates to the _print.scss that fixed the printed page margin and pass the lint test. I tried Atom and the scss lint, however I was unable to find the lint rules file (the readme github link doesn't work) and it seems by default the rules are different than our project's rules.

**Comment by jkleinsc** [comment]
@dapierce Looks good to me.  I'm going to merge this PR in.

**Comment by jglovier** [comment]
@dapierce oooff. My bad. Those docs need updated. We're using Stylelint now, not SCSS Lint anymore. 😖

**Pull request merged**